contract.md

```diff
@@ -12,11 +16,19 @@ like to work on, as part of your project (for example, other things related to D
 you have come across, or topics from other classes)? Why are you interested in working on
 these things?_
 
+I would like to work on external DSLs. While we have covered these conceptually in class, I found myself more challenged understanding how they work and what it would look like to implement one. Even for the PicoBot assignment, I experienced many difficulties so I felt pressured by time to use an internal DSL as that is what I felt I had a clearer understanding on and that I could confidently work with. 
+
+Thus, I am somewhat biased towards challenging myself to take on an external DSL approach for this project so that I can ensure I understand what that implementation entails. Since I had difficulties understanding how external DSLs work, I think that by implementing one I will be able to really clear up my understanding of external DSLs. However, I will first need to review the lessons on external DSLs, such as "Programs, interpreters, and compilers", to clear up any background information I will need going into beginning my implementation.
+
```

Comment:
I like that you want to challenge yourself this will be a great project for you.